great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -709,12 +724,14 @@ class SQLDatasource(Datasource):
     # private attrs
     _cached_connection_string: str = pydantic.PrivateAttr("")
     _engine: Union[sqlalchemy.engine.Engine, None] = pydantic.PrivateAttr(None)
+    # These are instance var because ClassVars can't contain Type variables. See
+    # https://peps.python.org/pep-0526/#class-and-instance-variable-annotations
+    _TableAsset: Type[TableAsset] = pydantic.PrivateAttr(TableAsset)
+    _QueryAsset: Type[QueryAsset] = pydantic.PrivateAttr(QueryAsset)
```

Comment:
Why set this here? I see it being used down below, but I don't understand why.

Reply:
This is a point of extension for subclasses and let's us decouple the creation of a [Table|Query]Asset from the datasource. That way sql specific engines can set these to construct their particular versions of the asset. See the sqlite datasource to see this in action.

Alternatively, I could have defined 2 methods `create_table_asset` and `create_query_asset` that the subclass could override and I could call these where I instantiate the asset.